modules/account-lib/src/coin/dot/transaction.ts

```diff
@@ -352,10 +353,29 @@ export class Transaction extends BaseTransaction {
     }
   }
 
+  constructSignedPayload(signature: Buffer): void {
+    const edSignature = `0x00${signature.toString('hex')}` as HexString;
+
+    this._signedTransaction = construct.signedTx(this._dotTransaction, edSignature, {
+      registry: this._registry,
+      metadataRpc: this._dotTransaction.metadataRpc,
+    });
+
+    this._signatures = [signature.toString('hex')];
+  }
+
   setTransaction(tx: UnsignedTransaction): void {
     this._dotTransaction = tx;
   }
 
+  /** @inheritdoc **/
+  get signablePayload(): Buffer {
```

Comment:
This is what you should get from `toBroadcastFormat()` but in string hex. How is this different?

Reply:
toBroadcastFormat is actually a serialized signature or txHex.

Signable payload is portion that actually needs to be signed. This is what the HSM will sign.